content/fundamental-redirects.js

```diff
@@ -0,0 +1,1145 @@
+  // BananaBread demo (bug 1238041)
+  // RewriteRule ^(\w{2,3}(?:-\w{2})?/)?demos/detail/bananabread$
+  // https://github.com/kripken/BananaBread/ [R=301,L]
+  localeRedirect(
+    /^demos\/detail\/bananabread$/i,
+    "https://github.com/kripken/BananaBread/",
+    { prependLocale: false, permanent: true }
+  ),
+  // RewriteRule ^(\w{2,3}(?:-\w{2})?/)?demos/detail/bananabread/launch$
+  // https://kripken.github.io/BananaBread/cube2/index.html [R=301,L]
+  localeRedirect(
+    /^demos\/detail\/bananabread\/launch$/i,
+    "https://kripken.github.io/BananaBread/cube2/index.html",
+    { prependLocale: false, permanent: true }
+  ),
+  // All other Demo Studio and Dev Derby paths (bug 1238037)
+  // RewriteRule ^(\w{2,3}(?:-\w{2})?/)?demos
+  // /$1docs/Web/Demos_of_open_web_technologies? [R=301,L]
+  localeRedirect(/^demos/i, "/docs/Web/Demos_of_open_web_technologies", {
+    permanent: true,
+  }),
+  // Legacy off-site redirects (bug 1362438)
+  // RewriteRule ^contests/ http://www.mozillalabs.com/ [R=302,L]
+  redirect(/^contests/i, "http://www.mozillalabs.com/", { permanent: false }),
+  // RewriteRule ^es4 http://www.ecma-international.org/memento/TC39.htm [R=302,L]
+  redirect(/^es4/i, "http://www.ecma-international.org/memento/TC39.htm", {
+    permanent: false,
+  }),
+];
+
+const zoneRedirects = [
+  [
+    "Add-ons",
+    "Mozilla/Add-ons",
+    [
+      "ar",
+      "bn",
+      "ca",
+      "de",
+      "en-US",
+      "es",
+      "fa",
+      "fr",
+      "hu",
+      "id",
+      "it",
+      "ja",
+      "ms",
+      "nl",
+      "pl",
+      "pt-BR",
+      "pt-PT",
+      "ru",
+      "sv-SE",
+      "th",
+      "uk",
+      "vi",
+      "zh-CN",
+      "zh-TW",
+      null,
+    ],
+  ],
+  ["Add-ons", "Mozilla/Πρόσθετα", ["el"]],
+  ["Add-ons", "Mozilla/애드온들", ["ko"]],
+  ["Add-ons", "Mozilla/Eklentiler", ["tr"]],
+  [
+    "Firefox",
+    "Mozilla/Firefox",
+    [
+      "ar",
+      "bm",
+      "ca",
+      "de",
+      "el",
+      "en-US",
+      "es",
+      "fi",
+      "fr",
+      "he",
+      "hi-IN",
+      "hu",
+      "id",
+      "it",
+      "ja",
+      "ko",
+      "ms",
+      "my",
+      "nl",
+      "pl",
+      "pt-BR",
+      "pt-PT",
+      "ru",
+      "sv-SE",
+      "th",
+      "tr",
+      "vi",
+      "zh-CN",
+      "zh-TW",
+      null,
+    ],
+  ],
+  ["Firefox", "Mozilla/ফায়ারফক্স", ["bn"]],
+  [
+    "Apps",
+    "Web/Apps",
+    ["en-US", "fa", "fr", "ja", "th", "zh-CN", "zh-TW", null],
+  ],
+  ["Apps", "Web/Aplicaciones", ["es"]],
+  ["Apps", "Apps", ["bn", "de", "it", "ko", "pt-Br", "ru"]],
+  ["Learn", "Learn", ["ca", "de", null]],
+  ["Apprendre", "Apprendre", ["fr"]],
+  [
+    "Marketplace",
+    "Mozilla/Marketplace",
+    ["de", "en-US", "es", "fr", "it", "ja", "zh-CN", null],
+  ],
+  ["Marketplace", "Mozilla/بازار", ["fa"]],
+];
+
+const zonePatternFmt = (prefix, zoneRootPattern) =>
+  new RegExp(`^${prefix}${zoneRootPattern}(?:\\/?|(?<subPath>[\\/$].+))$`);
```

Comment:
I'm realizing just now that all of the Regexp patterns in this file should be case-insensitive.
```suggestion
  new RegExp(`^${prefix}${zoneRootPattern}(?:\\/?|(?<subPath>[\\/$].+))$`, "i");
```

Reply:
I thought there were other patterns that were not yet case insensitive, but I think this is the only one.